### DISCUSSION TITLE
Error: Generated engines use disallowed import "node:stream". Only allowed imports are to Node.js builtin modules or e   ngines internals.

### DISCUSSION THREAD

**Discussion by runsense** [gatsbyjs/gatsby#39015] [Umbrella Discussions]
hello, 
in cross plateform project with capacitor, 
when i implement gatsby-ssr wrapRootElement for authorProvider
work on develop but
throw me this error when i build
i remake all before from 
rendering starter for catch error
package.json 
{
  "name": "gatsby-starter-rendering-modes",
  "private": true,
  "version": "1.0.0",
  "description": "A Gatsby starter to demonstrate new rendering modes in Gatsby 4",
  "main": "index.js",
  "scripts": {
    "start": "npm run develop",
    "develop": "gatsby develop",
    "build": "gatsby build",
    "serve": "gatsby serve",
    "clean": "gatsby clean"
  },
  "author": "Brent Jackson <jxnblk@gmail.com>",
  "license": "0BSD",
  "dependencies": {
    "@capacitor/android": "^6.1.0",
    "@capacitor/cli": "^6.1.0",
    "@capacitor/core": "^6.1.0",
    "@capacitor/ios": "^6.1.0",
    "firebase": "^10.12.2",
    "framework7": "^8.3.3",
    "framework7-react": "^8.3.3",
    "gatsby": "^5.0.0",
    "gatsby-plugin-breakpoints": "^1.3.11",
    "gatsby-plugin-firebase-v9.0": "^1.0.3",
    "gatsby-plugin-image": "^3.13.1",
    "gatsby-plugin-sharp": "^5.13.1",
    "gatsby-source-filesystem": "^5.13.1",
    "gatsby-source-strapi": "^3.3.3",
    "gatsby-transformer-remark": "^6.13.1",
    "gatsby-transformer-sharp": "^5.13.1",
    "node-fetch": "^2.6.2",
    "peerjs": "^1.5.4",
    "prop-types": "^15.8.1",
    "react": "^18.0.0",
    "react-dom": "^18.0.0"
  }
}
and global error
 Error: Generated engines use disallowed import "node:stream". Only allowed imports are to Node.js builtin modules or e
  ngines internals.

  - child.ts:60 Function._module.default._load
    [gatsby-starter-rendering-modes]/[gatsby]/src/utils/validate-engines/child.ts:60:11

  - loader:1233 Module.require
    node:internal/modules/cjs/loader:1233:19

  - helpers:179 require
    node:internal/modules/helpers:179:18

  - external node-commonjs "node:stream":1 Object.7075
    C:/Users/parisjc/Documents/react/gatsby-starter-rendering-modes/.cache/page-ssr/routes/webpack:/gatsby-starter-rende
    ring-modes/external node-commonjs "node:stream":1:1

  - bootstrap:19 __webpack_require__
    C:/Users/parisjc/Documents/react/gatsby-starter-rendering-modes/.cache/page-ssr/routes/webpack:/gatsby-starter-rende
    ring-modes/webpack/bootstrap:19:1

  - main.js:3 Object.8570
    [gatsby-starter-rendering-modes]/[@fastify]/busboy/lib/main.js:3:24

  - bootstrap:19 __webpack_require__
    C:/Users/parisjc/Documents/react/gatsby-starter-rendering-modes/.cache/page-ssr/routes/webpack:/gatsby-starter-rende
    ring-modes/webpack/bootstrap:19:1

  - body.js:3 Object.6628
    [gatsby-starter-rendering-modes]/[undici]/lib/fetch/body.js:3:16

  - bootstrap:19 __webpack_require__
    C:/Users/parisjc/Documents/react/gatsby-starter-rendering-modes/.cache/page-ssr/routes/webpack:/gatsby-starter-rende
    ring-modes/webpack/bootstrap:19:1

  - response.js:4 Object.2675
    [gatsby-starter-rendering-modes]/[undici]/lib/fetch/response.js:4:47

  - bootstrap:19 __webpack_require__
    C:/Users/parisjc/Documents/react/gatsby-starter-rendering-modes/.cache/page-ssr/routes/webpack:/gatsby-starter-rende
    ring-modes/webpack/bootstrap:19:1

  - index.js:119 Object.3477
    [gatsby-starter-rendering-modes]/[undici]/index.js:119:3

  - bootstrap:19 __webpack_require__
    C:/Users/parisjc/Documents/react/gatsby-starter-rendering-modes/.cache/page-ssr/routes/webpack:/gatsby-starter-rende
    ring-modes/webpack/bootstrap:19:1

  - render-page.js:10683 Module.9149
    C:/Users/parisjc/Documents/react/gatsby-starter-rendering-modes/.cache/page-ssr/routes/render-page.js:10683:14

  - bootstrap:19 require
    C:/Users/parisjc/Documents/react/gatsby-starter-rendering-modes/.cache/page-ssr/routes/webpack:/gatsby-starter-rende
    ring-modes/webpack/bootstrap:19:1

  - gatsby-ssr.js:12 Object.5328
    C:/Users/parisjc/Documents/react/gatsby-starter-rendering-modes/.cache/page-ssr/routes/webpack:/gatsby-starter-rende
    ring-modes/gatsby-ssr.js:12:24

**Comment by stbly** [comment]
I had a similar issue. I traced it to using wildcard imports like so:

`import * as utils from './utils`

Once I removed those imports my build compiled just fine.
